golang/btcspv/bitcoin_spv_test.go

```diff
@@ -100,8 +121,14 @@ func preprocessObject(m map[string]interface{}) {
 			l := v.([]interface{})
 			preprocessList(l)
 		case string:
-			// overwrite the string with a []byte
-			m[k] = DecodeIfHex(v.(string))
+			buf := DecodeIfHex(v.(string))
+			if len(buf) == 32 {
+				m[k], _ = NewHash256Digest(buf)
+			} else if len(buf) == 80 {
+				m[k], _ = NewRawHeader(buf)
+			} else {
+				m[k] = buf
+			}
```

Comment:
I noticed this block of code is repeated a lot, could it be extracted into another function?